src/devices/Mcp23xxx/Mcp23S17SpiAdapter.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Spi;
+
+namespace Iot.Device.Mcp23xxx
+{
+    public partial class Mcp23s17
+    {
+        /// <summary>
+        /// SPI adapter for Mcp23S17
+        /// </summary>
+        protected class Mcp23S17SpiAdapter : SpiAdapter
```

Comment:
can we fix old adapter to work here? what's the diff? (hard to see in here)

Reply:
I made a new adapter, because I don't know if the required modifications for this Class will broke the other classes or not.
I can only test the Mcp23S17 implementation, because I only have this hardware.

Reply:
I changed this
SpiAdapter.cs line 84:
```csharp
Span<byte> readBuffer = stackalloc byte[buffer.Length + 2];
```

To this
Mcp23s17.Mcp23S17SpiAdapter.cs line 42:
```csharp
Span<byte> readBuffer = stackalloc byte[
    writeBuffer.Length > buffer.Length
        ? writeBuffer.Length
        : buffer.Length
];
```